lib/bolt/inventory/inventory2.rb

```diff
@@ -0,0 +1,264 @@
+# frozen_string_literal: true
+
+require 'bolt/inventory/group2'
+
+module Bolt
+  class Inventory
+    class Inventory2
+      def initialize(data, config = nil, target_vars: {}, target_facts: {}, target_features: {})
+        @logger = Logging.logger[self]
+        # Config is saved to add config options to targets
+        @config = config || Bolt::Config.default
+        @data = data ||= {}
+        @groups = Group2.new(data.merge('name' => 'all'))
```

Comment:
Is the only diff `Group` to `Group2`?  Would `super`, then resetting `@groups` to Group2 work?

Reply:
If we hadn't decided to copy/paste that would have worked but processing all the groups is really expensive so we wouldn't want to do it twice. We could have created a load_groups method to capture this difference though.

As it is `super` would just call the base `object` initialize and won't work.